FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Common/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -58,19 +59,33 @@ public static void ValidateConfiguration(this IServiceCollection services)
                 throw new ConfigurationErrorException("Failed to parse job configuration", ex);
             }
 
-            if (string.IsNullOrEmpty(jobConfiguration.ContainerName))
+            ValidateAgentName(jobConfiguration.AgentName);
+
+            if (!Enum.IsDefined(typeof(QueueType), jobConfiguration.QueueType))
             {
-                throw new ConfigurationErrorException($"Target azure container name can not be empty.");
+                throw new ConfigurationErrorException(
+                    $"Queue type '{jobConfiguration.QueueType}' is not supported.");
             }
 
-            // TODO: add more validation for agent name, table url, queue url
-            // TODO: enable it when generic task is enabled
-            /*
-            if (string.IsNullOrEmpty(jobConfiguration.AgentName))
+            if (string.IsNullOrEmpty(jobConfiguration.TableUrl))
```

Comment:
Nit: Maybe **IsNullOrWhiteSpace**, also for below conditions.

Reply:
Fixed